src/Network/Datadog/Downtime.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-|
+Downtime prevents all alerting related to specific Datadog scopes.
+-}
+module Network.Datadog.Downtime
+( DowntimeSpec(..)
+, Downtime
+, dId
+, dSpec
+, DowntimeId
+, minimalDowntimeSpec
+, scheduleDowntime
+, updateDowntime
+, cancelDowntime
+, loadDowntime
+, loadDowntimes
+) where
+
+
+import Control.Monad (void)
+
+import Data.Aeson
+import Data.Aeson.Types (modifyFailure, typeMismatch)
+import qualified Data.HashMap.Strict as Data.HashMap (union)
+import Data.Text (Text)
+import Data.Time.Clock
+import Data.Time.Clock.POSIX
+import qualified Data.Vector (head)
+
+import Network.Datadog
+import Network.Datadog.Internal
+
+
+-- | A description of when downtime should occur.
+data DowntimeSpec = DowntimeSpec { dsScope :: Tag
+                                   -- ^ The scope to apply downtime to (if applying downtime to a
+                                   -- host, use a tag of the form "host:hostname", NOT just
+                                   -- "hostname")
+                                 , dsStart :: Maybe UTCTime
+                                   -- ^ When to start the downtime (or immediately)
+                                 , dsEnd :: Maybe UTCTime
+                                   -- ^ When to stop the downtime (or indefinitely)
+                                 , dsMessage :: Maybe Text
+                                   -- ^ A message to include with notifications for this downtime
+                                 } deriving (Eq)
+
+instance ToJSON DowntimeSpec where
+  toJSON ds = object (["scope" .= dsScope ds]
+                      ++ maybe [] (\a -> ["start" .= (ceiling (utcTimeToPOSIXSeconds a) :: Integer)]) (dsStart ds)
+                      ++ maybe [] (\a -> ["end" .= (floor (utcTimeToPOSIXSeconds a) :: Integer)]) (dsEnd ds)
+                      ++ maybe [] (\a -> ["message" .= a]) (dsMessage ds)
+                     )
+
+instance FromJSON DowntimeSpec where
+  parseJSON (Object v) = modifyFailure ("DowntimeSpec: " ++) $
+                         DowntimeSpec <$>
+                         (withArray "Text" (parseJSON . Data.Vector.head) =<< v .: "scope") <*>
```

Comment:
I'd prefer to avoid using `Data.Vector.Head` here since it's not a total function.